### ISSUE TITLE
Google Calendar desklet doesn't show today/current date and can't select calendars

### ISSUE THREAD

**Issue by Auerilas** [linuxmint/cinnamon-spices-desklets#330]
Is there any way to get the Google Calendar desklet to show "Today" in the agenda? It shows up in gcalcli when i use galcli agenda.

Also, perhaps a related or unrelated issue, but choosing the calendar doesn't seem to work. If I choose 'Home', 'Research', or 'Teaching', it always shows results from all the calendars and doesn't select specific ones. Here are screenshots to compare gcalcli and the desklet for selecting calendars.

![research](https://user-images.githubusercontent.com/9372234/43530520-05ce3d86-956b-11e8-90be-a8cfb91ca3d5.png)
![home](https://user-images.githubusercontent.com/9372234/43530529-087147d6-956b-11e8-96a2-320b0d0b4068.png)
![teaching](https://user-images.githubusercontent.com/9372234/43530534-0c3db070-956b-11e8-9fd7-85cd579030f5.png)

Looks like there are some desklet errors when it loads. Maybe related?

```
 * Desklet Google Calendar 07/08/2018 
* Cinnamon version 3.8.8
 * Mint 19
 * ~/.xsession-errors
(nm-applet:1696): Gtk-WARNING **: 08:56:42.416: Can't set a parent on widget which has a parent

(nm-applet:1696): Gtk-CRITICAL **: 08:56:42.445: gtk_widget_destroy: assertion 'GTK_IS_WIDGET (widget)' failed

(nm-applet:1696): Gtk-CRITICAL **: 08:56:42.445: gtk_widget_destroy: assertion 'GTK_IS_WIDGET (widget)' failed

(nm-applet:1696): Gtk-WARNING **: 08:56:42.468: Can't set a parent on widget which has a parent
Cjs-Message: 08:56:42.496: JS LOG: About to start Cinnamon
** Message: 08:56:42.550: nemo-desktop: session is cinnamon, establishing proxy
Cjs-Message: 08:56:42.589: JS LOG: [LookingGlass/info] Cinnamon.AppSystem.get_default() started in 90 ms
Cjs-Message: 08:56:42.595: JS LOG: [LookingGlass/info] loading user theme: /usr/share/themes/Mint-Y-Dark/cinnamon/cinnamon.css
Cjs-Message: 08:56:42.610: JS LOG: [LookingGlass/info] added icon directory: /usr/share/themes/Mint-Y-Dark/cinnamon
Cjs-Message: 08:56:42.693: JS LOG: [LookingGlass/info] PlacesManager: Updating devices
St-Message: 08:56:42.696: cogl npot texture sizes SUPPORTED
Cjs-Message: 08:56:42.706: JS LOG: [LookingGlass/info] loaded at Tue Jul 31 2018 08:56:42 GMT-0600 (MDT)
Cjs-Message: 08:56:42.706: JS LOG: Cinnamon started at Tue Jul 31 2018 08:56:42 GMT-0600 (MDT)
Cjs-Message: 08:56:42.731: JS LOG: [LookingGlass/info] ExtensionSystem started in 0 ms
Cjs-Message: 08:56:42.731: JS LOG: [LookingGlass/info] SearchProviderManager started in 0 ms
openGL version 4.5 detected (GL3 Cogl Driver)
MetaSyncRing disabled: couldn't find required GL extensions, or the minimum safe openGL version was not met
Cjs-Message: 08:56:42.932: JS LOG: [LookingGlass/info] Loaded applet show-desktop@cinnamon.org in 40 ms
Cjs-Message: 08:56:42.957: JS LOG: [LookingGlass/info] Loaded applet user@cinnamon.org in 24 ms
Cjs-Message: 08:56:42.983: JS LOG: [LookingGlass/info] Loaded applet calendar@cinnamon.org in 26 ms
Cjs-Message: 08:56:42.990: JS LOG: [LookingGlass/info] Loaded applet removable-drives@cinnamon.org in 8 ms
Cjs-Message: 08:56:43.008: JS LOG: [LookingGlass/info] Loaded applet systray@cinnamon.org in 18 ms
Cjs-Message: 08:56:43.013: JS WARNING: [/home/nate/.local/share/cinnamon/desklets/googleCalendar@javahelps.com/lib/xdate.js 13]: test for equality (==) mistyped as assignment (=)?

(cinnamon:1652): Cjs-WARNING **: 08:56:43.015: Some code accessed the property 'XDate' on the module 'utility'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
(cinnamon:1652): Cjs-WARNING **: 08:56:43.015: Some code accessed the property 'SpawnReader' on the module 'utility'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.

(cinnamon:1652): Cjs-WARNING **: 08:56:43.015: Some code accessed the property 'CalendarUtility' on the module 'utility'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.

Cjs-Message: 08:56:43.021: JS LOG: [LookingGlass/info] Role locked: notifications

Cjs-Message: 08:56:43.027: JS LOG: [LookingGlass/info] Loaded applet notifications@cinnamon.org in 18 ms

Cjs-Message: 08:56:43.043: JS LOG: [LookingGlass/info] Loaded applet keyboard@cinnamon.org in 16 ms

Cjs-Message: 08:56:43.045: JS WARNING: [/usr/share/cinnamon/js/misc/fileUtils.js line 210 > Function 69]: reference to undefined property "eventsList"

Cjs-Message: 08:56:43.047: JS WARNING: [/home/nate/.local/share/cinnamon/desklets/googleCalendar@javahelps.com/lib/xdate.js 13]: reference to undefined property "-1"

Cjs-Message: 08:56:43.052: JS WARNING: [/home/nate/.local/share/cinnamon/desklets/googleCalendar@javahelps.com/lib/utility.js 50]: assignment to undeclared variable res

Cjs-Message: 08:56:43.057: JS LOG: [LookingGlass/info] Loaded desklet googleCalendar@javahelps.com in 14 ms
 ```

**Notify author of desklet**
@schorschii

**Comment by jaszhix** [comment]
Author of desklet is actually @slgobinath

**Comment by Auerilas** [comment]
Interestingly, the "TODAY" issue appears on both of my machines. The "Calendar selection" issue only appears on my laptop. Calendar selection works fine on my desktop. The only difference I can think of is that I installed gcalcli after the desklet on my laptop and before the desklet on my desktop. I'll try a reinstall of the desklet on my laptop to see if that resolves the issue.

**Comment by Auerilas** [comment]
After even more inspection, I dug into the desklet.js script. I've never seen or written javascript, so I can't understand 90% of it. But I did find this:

```
getCalendarCommand() {
        let dateTime = new Date();
        let command = ["gcalcli"];
        if (this.calendarName != "") {
            let calendars = this.calendarName.split(",");
            for (let name of calendars) {
                name = name.trim();
                if (name !== "") {
                    command.push("--calendar");
                    command.push(name);
                }
            }
        }
        command.push("agenda");
        command.push(CalendarUtility.formatParameterDate(dateTime));
        if (this.interval == null) {
            this.interval = 7; // Default interval is 7 days
        }
        dateTime.setDate(dateTime.getDate() + this.interval);
        command.push(CalendarUtility.formatParameterDate(dateTime));
        command.push("--nostarted");
        command.push("--tsv");     
        return command;
    },
```
which I assumed were the instructions to gcalcli. In particular, this option: `command.push("--nostarted");` tells gcalcli not to return events that have already started. Since I don't have a better solution to separate 'all day' events from scheduled events that have occurred, I commented this out and now my agenda shows all events for the day. Which is fine by me. Maybe in the future, it'd be possible to separate out all day vs. prior scheduled events, but for now I suggest a configuration option that toggles this command on or off: like a "show all events for the day" or "show started events" button.

**Comment by slgobinath** [comment]
Hi,
> A reinstall fixed the calendar selection issue

Thanks for letting me know this. I also have observed the problem with `TODAY`. It is actually a problem in `gcalcli` side.

If `--nostarted` is used with `gcalcli v3.3.2`, it will ignore events which are already started except full-day events like Birthdays. However, in the latest version of gcalcli, `--nostarted`  even ignores full-day events. I am also affected by this issue on my LM 19 (but not on LM 18).

I am planning to remove this argument and to filter events within the desklet as a fix. However, I am a little busy with my thesis these days. I will fix this problem once I get some time.

**Comment by Auerilas** [comment]
No worries! Thanks for looking into it. In the meantime, I've raised an issue on gcalcli, as this seems like an issue. If its fixed on the gcalcli side, then that should fix the issue here, as well.

thanks again!

**Comment by slgobinath** [comment]
Hi @Auerilas,
This problem will be fixed in #342. Could you please help me by testing this PR in your system because it contains a major change in the code which may break the desklet.

Thanks

**Comment by slgobinath** [comment]
Hi @Auerilas,

Please note that this issue has been fixed with the recent PR. Now we do not rely on `gcalcli`, Please note that you need to install some new dependencies to get the latest desklet to work.

@jaszhix you can close this issue after the confirmation.

Thanks
